docs/DeveloperGuide.md

```diff
@@ -1,38 +1,135 @@
+
+### Value Proposition
+SpendSwift provides a fast, text-based solution for managing finances, eliminating the complexity of traditional budgeting tools.
 
 ## User Stories
-
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+<!-- @@author glenda-1506 -->
+| Version | As a... | I want to... | So that I can... |
+|---------|---------|--------------|------------------|
+| v1.0    | Budget-conscious user | Quickly log an expense using a typed command (e.g., add 50 groceries) | track my spending with easy input |
+| v1.0    | Budget-conscious user | View my budget for all categories | see how much I could spend |
+| v1.0    | Budget-conscious user | Set a monthly reset for my budget tracking | start each month fresh with my budgeting |
+| v1.0    | Budget planner | View all my expenses | monitor what I have been spending on |
+| v1.0    | Frequent user | Set a budget limit for each category (e.g., set budget 200 groceries) | limit my spending according to categories |
+| v1.0    | Frequent user | Delete an expense entry (e.g., delete 5) | quickly correct mistakes |
+| v1.0    | Frequent user | Categorize expenses (e.g., add category food) | customize my expense tracking to better manage my budget |
 
 ## Non-Functional Requirements
-
-{Give non-functional requirements}
+<!-- @@author mayfairmi6 -->
+| ID  | Requirement      | Description                                          | Rationale                                             |
+|-----|------------------|------------------------------------------------------|-------------------------------------------------------|
+| 1   | Responsiveness   | The system should respond to user commands within 2 seconds. | Ensures efficient interaction and enhances user satisfaction. |
+| 2   | Data Integrity   | The system must maintain accurate tracking and updating of financial entries. | Prevents discrepancies in financial reporting, ensuring reliability. |
+| 3   | User Error Handling | The system should provide clear error messages and support easy correction of user inputs. | Facilitates management of entries and reduces user frustration. |
+| 4   | Customizability  | Users should be able to easily add and modify expense categories. | Allows users to tailor the system to their specific needs. |
+| 5   | Automated Tasks  | Support automated budget resets at the start of each month. | Minimizes user effort in maintaining accurate monthly tracking. |
+| 6   | Accessibility    | The chat interface should be simple and intuitive.   | Ensures that all users can effectively interact with the system without extensive training. |
```

Comment:
Similarly, the formatting of this is off. 
![image](https://github.com/user-attachments/assets/4c640b08-1538-4478-a3b8-8feecc914558)